wait/host_port.go

```diff
@@ -28,9 +28,15 @@ type HostPortStrategy struct {
 	// all WaitStrategies should have a startupTimeout to avoid waiting infinitely
 	timeout      *time.Duration
 	PollInterval time.Duration
+
+	// SkipInternalCheck is a flag to skip the internal check, which is useful when
+	// a shell is not available in the container or when the container doesn't bind
+	// the port internally until additional conditions are met.
+	SkipInternalCheck bool
```

Comment:
NIT: do we neet an exposed field here? Not a blocker though

Reply:
Nope was just following the example from Port and PollInterval, I guess that's something to review in v1 too